docs/guides/understand_logs_and_metrics.md

```diff
@@ -0,0 +1,252 @@
+# Understand Logs and Metrics
+
+## Introduction
+
+When you run a training job, MaxText produces detailed output logs. This guide shows you how to interpret these logs to understand your configuration and monitor performance.
+
+To start, run a simple pretraining job on a single-host TPU. The logs from this command will be used as an example throughout this guide.
+
+```bash
+python3 -m MaxText.train MaxText/configs/base.yml \
+base_output_directory=gs://runner-maxtext-logs run_name=demo \
+model_name=deepseek2-16b \
+per_device_batch_size=1 max_target_length=2048 steps=10 dataset_type=synthetic enable_checkpointing=false
+```
+
+## 1 Configuration Info
+
+The first section of the log details the final configuration of your run. This is crucial for debugging, as it shows you exactly which parameters were used. 
+
+MaxText builds its configuration in layers. 
+- It starts with the **default configuration** from the YAML file [base.yml](https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/configs/base.yml), as you provide `MaxText/configs/base.yml`. 
+
+- Then, it overwrites any of these values with the arguments you provide in the **command line**.
+  ```
+  Updating keys from env and command line: ['run_name', 'model_name', 'enable_checkpointing', 'base_output_directory', 'per_device_batch_size', 'dataset_type', 'steps', 'max_target_length']
+  ```
+- It updates parameters based on the **model-specific configuration** file. As you specify `model_name=deepseek2-16b`, it read from [deepseek2-16b.yml](https://github.com/AI-Hypercomputer/maxtext/blob/main/MaxText/configs/models/deepseek2-16b.yml).
+
+  ```
+  Running Model: deepseek2-16b
+  Updating following parameters in config
+
+  base_emb_dim: 2048
+  base_num_query_heads: 16
+  ...
+  Updating keys from model: ['base_emb_dim', 'base_num_query_heads', 'base_num_kv_heads', 'base_mlp_dim', 'base_moe_mlp_dim', 'base_num_decoder_layers', 'first_num_dense_layers', 'mlp_activations', 'vocab_size', 'enable_dropout', 'logits_via_embedding', 'normalization_layer_epsilon', 'num_experts', 'num_experts_per_tok', 'shared_experts', 'routed_scaling_factor', 'routed_score_func', 'routed_bias', 'decoder_block', 'attention_type', 'q_lora_rank', 'kv_lora_rank', 'qk_nope_head_dim', 'qk_rope_head_dim', 'v_head_dim', 'rope_type', 'rope_max_timescale', 'max_position_embeddings', 'original_max_position_embeddings', 'rope_factor', 'beta_fast', 'mscale']
+  ```
+
+The final, consolidated configuration is printed last.
```

Comment:
Probably worth to mention that you cannot modify a config from both model and cli: https://github.com/AI-Hypercomputer/maxtext/blob/eff346c028092c4f4fd421e5c5343308def5de5a/MaxText/pyconfig.py#L461